examples/icd_code_prediction/icd_coding.py

```diff
@@ -0,0 +1,107 @@
+import sys
+import yaml
+from termcolor import colored
+
+from forte.common.configuration import Config
+from forte.data.data_pack import DataPack
+from forte.data.readers import PlainTextReader
+from forte.pipeline import Pipeline
+from forte.processors.writers import PackIdJsonPackWriter
+
+from ft.onto.base_ontology import (
+    Token,
+)
+from fortex.spacy import SpacyProcessor
+
+# import MedicalArticle to get article data from data pack
+from ftx.medical.clinical_ontology import MedicalArticle
+
+# import mimic3 dataset
+from fortex.health.readers.mimic3_note_reader import Mimic3DischargeNoteReader
+
+# import ICD coding processor
+from fortex.health.processors.icd_coding_processor import (
+    ICDCodingProcessor,
+)
+
+def main(
+    input_path: str, # Path to mimic3 data if use_mimic3_reader=True else path to note directory
+    output_path: str, # Path to output directory
+    max_packs: int = -1, # Max number of notes to read from mimic3 dataset. Set to -1 to read all.
+    use_mimic3_reader: bool = True, # Read from mimic3 dataset or plain text
+):
+    # create a pipeline
+    pl = Pipeline[DataPack]()
+
+    # select reader: read from mimic3 dataset or from files
+    if use_mimic3_reader is False:
+        pl.set_reader(PlainTextReader())
+    else:
+        pl.set_reader(
+            Mimic3DischargeNoteReader(), config={"max_num_notes": max_packs}
+        )
+
+    # load config
+    config = Config(yaml.safe_load(open("config.yml", "r")), None)
+
+    # add processors
+    # use spacy for tokenizing
+    pl.add(SpacyProcessor(), config.Spacy)
+    # predict icd code
+    pl.add(ICDCodingProcessor(), config.ICD)
+    # add writer to save the output to the disk
```

Comment:
I don't think we need comments for each line in the code, some lines are self explanatory. Maybe we can reduce the clutter by deleting unnecessary comments?